src/others/xml/lv_xml_component.c

```diff
@@ -0,0 +1,300 @@
+    state.ctx.name = name;
+
+    /* Parse the XML to extract metadata */
+    XML_Parser parser = XML_ParserCreate(NULL);
+    XML_SetUserData(parser, &state);
+    XML_SetElementHandler(parser, start_metadata_handler, end_metadata_handler);
+
+    if(XML_Parse(parser, xml_def, lv_strlen(xml_def), XML_TRUE) == XML_STATUS_ERROR) {
+        LV_LOG_WARN("XML parsing error: %s on line %lu",
+                    XML_ErrorString(XML_GetErrorCode(parser)),
+                    (unsigned long)XML_GetCurrentLineNumber(parser));
+        XML_ParserFree(parser);
+        return LV_RESULT_INVALID;
+    }
+
+    XML_ParserFree(parser);
+
+    /* Copy extracted metadata to component processor */
+    lv_xml_component_ctx_t * ctx = lv_ll_ins_head(&component_ctx_ll);
+    lv_memzero(ctx, sizeof(lv_xml_component_ctx_t));
+    lv_memcpy(ctx, &state.ctx, sizeof(lv_xml_component_ctx_t));
+
+    /* Extract view content directly instead of using XML parser */
+    ctx->view_def = extract_view_content(xml_def);
+    ctx->name = lv_strdup(name);
+    if(!ctx->view_def) {
+        LV_LOG_WARN("Failed to extract view content");
+        /* Clean up and return error */
+        lv_free(ctx);
+        return LV_RESULT_INVALID;
+    }
+
+    return LV_RESULT_OK;
+}
+
+
+lv_result_t lv_xml_component_register_from_file(const char * path)
+{
+    /* Extract component name from path */
+    /* Create a copy of the filename to modify */
+    char * filename = lv_strdup(lv_fs_get_last(path));
+    const char * ext = lv_fs_get_ext(filename);
+    filename[lv_strlen(filename) - lv_strlen(ext) - 1] = '\0'; /*Trim the extension*/
+
+    lv_fs_res_t fs_res;
+    lv_fs_file_t f;
+    fs_res = lv_fs_open(&f, path, LV_FS_MODE_RD);
+    if(fs_res != LV_FS_RES_OK) {
+        LV_LOG_WARN("Couldn't open %s", path);
+        lv_free(filename);
+        return LV_RESULT_INVALID;
+    }
+
+    /* Determine file size */
+    lv_fs_seek(&f, 0, LV_FS_SEEK_END);
+    uint32_t file_size = 0;
+    lv_fs_tell(&f, &file_size);
+    lv_fs_seek(&f, 0, LV_FS_SEEK_SET);
+
+    /* Create the buffer */
+    char * xml_buf = lv_malloc(file_size + 1);
+    if(xml_buf == NULL) {
+        LV_LOG_WARN("Memory allocation failed for file %s (%d bytes)", path, file_size + 1);
+        lv_free(filename);
+        lv_fs_close(&f);
+        return LV_RESULT_INVALID;
+    }
+
+    /* Read the file content  */
+    uint32_t rn;
+    lv_fs_read(&f, xml_buf, file_size, &rn);
+    if(rn != file_size) {
+        LV_LOG_WARN("Couldn't read %s fully", path);
+        lv_free(filename);
+        lv_free(xml_buf);
+        lv_fs_close(&f);
+        return LV_RESULT_INVALID;
+    }
+
+    /* Null-terminate the buffer */
+    xml_buf[rn] = '\0';
+
+    /* Register the component */
+    lv_result_t res = lv_xml_component_register_from_data(filename, xml_buf);
+
+    /* Housekeeping */
+    lv_free(filename);
+    lv_free(xml_buf);
+    lv_fs_close(&f);
+
+    return res;
+}
+
+void lv_xml_component_unregister(const char * name)
```

Comment:
Why not return lv_result_t since NULL means invalid parameter? It would be nice to track failures like this.

Reply:
Good idea, added.